File: src/main/java/com/lostfound/service/AuthService.java
package com.lostfound.service;

import model.User;
import org.springframework.stereotype.Service;

/**
 * AuthService acts as a thin wrapper around UserService.
 * Handles login, registration, and password reset logic for the UI layer.
 */
@Service
public class AuthService {

    private final UserService userService = new UserService();

    /**
     * Register a new user with default role = "USER".
     * SRM email restriction is enforced inside UserService.
     */
    public boolean register(String username, String plainPassword, String email) {
        return userService.registerUser(username, plainPassword, "USER", email);
    }

    /**
     * Authenticate a user by username and password.
     */
    public User login(String username, String plainPassword) {
        return userService.login(username, plainPassword);
    }

    // ---------------- Forgot Password Support ----------------

    /**
     * Initiate password reset: generates a token and saves it with expiry.
     * Returns the token if successful, null otherwise.
     */
    public String initiatePasswordReset(String email) {
        return userService.initiatePasswordReset(email);
    }

    /**
     * Validate a reset token.
     */
    public boolean validateResetToken(String token) {
        return userService.validateResetToken(token);
    }

    /**
     * Reset password using a valid token.
     */
    public boolean resetPassword(String token, String newPassword) {
        return userService.resetPassword(token, newPassword);
    }
}

File: frontend/src/services/authService.ts
import { get, post } from './api';
import { LoginResponse, User } from '../types';

export const authService = {
  async login(username: string, password: string): Promise<LoginResponse> {
    return post<LoginResponse>('/auth/login', { username, password });
  },

  async register(username: string, password: string, email: string, role: string = 'USER'): Promise<LoginResponse> {
    return post<LoginResponse>('/auth/register', { username, password, email, role });
  },

  async getCurrentUser(userId: number): Promise<{ success: boolean; user: User }> {
    return get<{ success: boolean; user: User }>(`/auth/me?userId=${userId}`);
  },

  async checkUsername(username: string): Promise<boolean> {
    const response = await get<{ exists: boolean }>(`/users/check-username?username=${username}`);
    return response.exists;
  },

  async checkEmail(email: string): Promise<boolean> {
    const response = await get<{ exists: boolean }>(`/users/check-email?email=${email}`);
    return response.exists;
  },

  async forgotPassword(email: string): Promise<{ success: boolean; message?: string; token?: string }> {
    return post<{ success: boolean; message?: string; token?: string }>(`/auth/forgot`, { email });
  },

  async resetPassword(token: string, newPassword: string): Promise<{ success: boolean; message?: string }> {
    return post<{ success: boolean; message?: string }>(`/auth/reset`, { token, newPassword });
  },
};


